Create rule S6562: Always set the DateTimeKind when creating a new DateTime object (#1712)
SonarSource/rspec: rules/S6562/whyAnIssue.adoc

> Not knowing what kind of date the application is using can lead to misunderstanding when printing or comparing dates. When there is a `DateTime` constructor with an overload that takes the same parameters and a `DateTimeKind`, the one with `DateTimeKind` should be used. This helps the application and its maintainers understand what kind of date is being managed. Most applications handle the same `DateTimeKind`, being intentional about it will reduce the chance of a later mismatch between `DateTime` isntances.

Reviewer: ```suggestion
Not knowing what is the `Kind` of the `DateTime` object that an application is using can lead to misunderstandings when displaying or comparing such objects. Explicitly setting the `Kind` property helps the application to stay consistent and its maintainers understand what kind of date is being managed. 
To achieve this, when there is a `DateTime` constructor with an overload that takes the same parameters and, additionally the `DateTimeKind`, you should always use it.
```

> Not knowing the `Kind` of the `DateTime` object that an application is using can lead to misunderstandings when displaying or comparing such objects. Explicitly setting the `Kind` property helps the application to stay consistent, and its maintainers understand what kind of date is being managed.

Reviewer: ```suggestion
Not knowing the `Kind` of the `DateTime` object that an application is using can lead to misunderstandings when displaying or comparing them. Explicitly setting the `Kind` property helps the application to stay consistent, and its maintainers understand what kind of date is being managed.
```

> To achieve this, when there is a `DateTime` constructor with an overload that takes the same parameters and, additionally, the `DateTimeKind`, you should always use it.

Reviewer: ```suggestion
To achieve this, when instantiating a new `DateTime` object you should always use aconstructor overload that allows you to define the `Kind` property.
```
